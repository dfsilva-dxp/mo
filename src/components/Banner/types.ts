export type BannerProps = {
  title: string;
  description: string;
  short_description: string;
  image_url: string;
};
