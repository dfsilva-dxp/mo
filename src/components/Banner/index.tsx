import { BannerProps } from "./types";

import {
  Description,
  Display,
  Picture,
  ShortDescription,
  Title,
  Wrapper
} from "./styles";

export default function Banner({
  title,
  description,
  short_description,
  image_url
}: BannerProps) {
  return (
    <Wrapper>
      <Display>
        <Title>{title}</Title>
        <Description>{description}</Description>
        <ShortDescription>{short_description}</ShortDescription>
      </Display>
      <Picture
        src={image_url}
        alt="Mulher segurando um cofrinho em formato de porco, pensando no seus gastos."
        title="Mulher segurando um cofrinho em formato de porco, pensando no seus gastos."
        role="img"
        aria-label="finance"
        loading="lazy"
      />
    </Wrapper>
  );
}
