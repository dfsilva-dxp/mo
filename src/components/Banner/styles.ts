import styled, { css } from "styled-components";

export const Wrapper = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: 87rem;
    height: 24rem;
    background: ${theme.colors.panel};
    padding: ${theme.spacings.xsmall};
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    place-items: center;
    border-radius: ${theme.border.radius};
  `}
`;

export const Display = styled.div`
  ${({ theme }) => css`
    display: grid;
    gap: ${theme.spacings.xxsmall};
  `}
`;

export const Title = styled.h4``;

export const Description = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xxlarge};
    line-height: 1;
    font-weight: ${theme.font.bold};
  `}
`;

export const ShortDescription = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.gray400};
    line-height: 1;
  `}
`;

export const Picture = styled.img``;
