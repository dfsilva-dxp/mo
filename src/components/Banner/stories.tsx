import { ComponentStory, ComponentMeta } from "@storybook/react";

import Banner from ".";

export default {
  title: "Header/Banner",
  component: Banner,
  argTypes: {},
  args: {
    title: "Bem-vindo (a) ao Mo.!",
    description: "NÃ³s te ajudamos a manter seus SONHOS...",
    short_description: "...e seu dinheiro querendo mais dinheiro!",
    image_url: "./img/personal-finance.svg"
  },
  parameters: {
    layout: "centered"
  },
  decorators: [
    (Story) => (
      <div style={{ width: "87rem", height: "24rem" }}>
        <Story />
      </div>
    )
  ]
} as ComponentMeta<typeof Banner>;

export const Basic: ComponentStory<typeof Banner> = ({ ...props }) => (
  <Banner {...props} />
);
