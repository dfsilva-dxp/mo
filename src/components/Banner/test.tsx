import { screen } from "@testing-library/react";
import { renderWithTheme } from "utils/tests/helpers";

import Banner from ".";

describe("Banner Component", () => {
  it("should render currectly banner", () => {
    const { container } = renderWithTheme(
      <Banner
        title="title"
        description="description"
        short_description="short description"
        image_url="./img/personal-finance.svg"
      />
    );

    expect(screen.getByRole("heading", { name: "title" })).toBeInTheDocument();
    expect(screen.getByText("description")).toBeInTheDocument();
    expect(screen.getByText("short description")).toBeInTheDocument();
    expect(screen.getByRole("img", { name: /finance/i })).toHaveAttribute(
      "src",
      "./img/personal-finance.svg"
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
