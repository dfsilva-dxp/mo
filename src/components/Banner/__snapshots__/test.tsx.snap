// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Component should render currectly banner 1`] = `
.c0 {
  width: 100%;
  max-width: 87rem;
  height: 24rem;
  background: #1F2029;
  padding: 1.6rem;
  display: grid;
  grid-template-columns: repeat(2,1fr);
  place-items: center;
  border-radius: 0.4rem;
}

.c1 {
  display: grid;
  gap: 0.8rem;
}

.c2 {
  font-size: 4.8rem;
  line-height: 1;
  font-weight: 700;
}

.c3 {
  color: #797D9A;
  line-height: 1;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h4
      class=""
    >
      title
    </h4>
    <p
      class="c2"
    >
      description
    </p>
    <p
      class="c3"
    >
      short description
    </p>
  </div>
  <img
    alt="Mulher segurando um cofrinho em formato de porco, pensando no seus gastos."
    aria-label="finance"
    class=""
    loading="lazy"
    role="img"
    src="./img/personal-finance.svg"
    title="Mulher segurando um cofrinho em formato de porco, pensando no seus gastos."
  />
</div>
`;
